altair/vegalite/v5/api.py

```diff
@@ -3813,29 +3813,14 @@ def add_selection(self, *params) -> Self:  # noqa: ANN002
         """'add_selection' is deprecated. Use 'add_params' instead."""
         return self.add_params(*params)
 
-    def interactive(  # noqa: C901
+    def interactive(
         self,
         name: str | None = None,
         bind_x: bool = True,
         bind_y: bool = True,
         tooltip: bool = True,
-        legend: bool
-        | Literal[
-            "color",
-            "fill",
-            "shape",
-            "stroke",
-            "opacity",
-            "fillOpacity",
-            "strokeOpacity",
-            "strokeWidth",
-            "strokeDash",
-            "angle",  # TODO Untested
-            "radius",  # TODO Untested
-            "radius2",  # TODO Untested
-            # "size",  # TODO Currently size is not working, renders empty legend
-        ] = True,
-    ) -> Self:
+        legend: bool | LegendChannel_T = False,
```

Comment:
Had to change the default to `False` (for now), as you'll need to adapt the existing tests to handle the new behaviour.